react-backend/routes/cbauthentication/checkCBlogin.js

```diff
@@ -0,0 +1,36 @@
+const validator = require('validator');
+
+const express = require('express');
+
+const hashCB = require('../../functions/cbhash');
+
+const router = express.Router();
+const getCBlogindetailsvalid = require('../../database/queries/getCBlogindetailsvalid');
+
+router.post('/', (req, res, next) => {
+  let body = '';
+  req.on('data', (chunk) => {
+    body += chunk;
+  });
+
+  req.on('end', () => {
+    const data = JSON.parse(body);
+    if (data.formEmail.length === 0 || data.formPswd.length === 0) {
+      res.send('noinput');
+    } else if (validator.isEmail(data.formEmail)) {
+      data.formPswd = hashCB(data.formPswd);
+      getCBlogindetailsvalid(data.formEmail, data.formPswd, (error, result) => {
+        if (error) {
+          console.log('error from getCBlogindetailsvalid ', error);
+        } else {
+          res.send(result.rows[0].exists);
+        }
+      });
+    } else if (!validator.isEmail(data.formEmail)) {
```

Comment:
this can just be an else as you will only ever hit it if `validator.isEmail(data.formEmail)` is falsey